### DISCUSSION TITLE
Regarding decomp_2d subroutines

### DISCUSSION THREAD

**Discussion by wwdhd** [xcompact3d/Incompact3d#271] [Q&A] [answered]
I tried to understand how the statistics are collected inside the statistics.f90. What I understand from the code is that they use and call decomp_2d_* a lot. But I do not find the original subroutines in the _src_ folder, Where is it and how does it do? Thanks a lot
Widhi

**Comment by slaizet** [comment, marked as answer]
Hi,
By default the statistics saved are u,v,w,p,u^2, v^2 and w^2. Please have a look at the videos on the xcompact3d readthedocs website to see how to post-process the data.
Best,
Sylvain
